packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1827,6 +1827,10 @@ export class ContainerRuntime
 			this.attachState !== AttachState.Attached || this.hasPendingMessages();
 		context.updateDirtyContainerState(this.dirtyContainer);
 
+		// Reference Sequence Number may have just changed, and it must be consistent across a batch, so flush now.
+		// This is coverage for old loaders that don't call ContainerRuntime.process for non-runtime messages.
+		this.deltaManager.on("op", () => this.flush());
```

Comment:
I would recommend adding a test that covers this scenario to prevent future regressions. There is a lot of churn in the op processing logic with more to come so a test will help.

Reply:
Yeah, you're right, I should do that.

Reply:
@markfields / @agarwal-navin 
Approving PR in case we wish to add tests as a follow up. Sooner check in will help me unblock my PR about enabling short ids by defaukt.